### DISCUSSION TITLE
How to insert and update element in array

### DISCUSSION THREAD

**Discussion by aakashkag** [prisma/prisma#2614] [Q&A] [answered]
Hi,
I am using Prisma2 with GraphQL. 
Database: Postgres

```sql
CREATE TABLE "Content" (
     "content_id" serial NOT NULL,
	"content_parent_id" int,
	"content_name" varchar(255) NOT NULL,
	"childs_id" integer[],
	"content_type" varchar(255) NOT NULL,
	CONSTRAINT "Content_pk" PRIMARY KEY ("content_id")
);
```

Currently while content creation I am able to create array using

```js
args.childs_id = { set: [1,2,3,4,5] };
```

Now lt's say I want to insert new element called 6. how we can do that.
what if we want to insert at any specific index of `array. childs_id[2] = 10`.
Or remove element from array.

I am struggling to understand how this operation we can perform in Prisma2

1)Insert
2)Insert at specific index
3)delete element

**Comment by ryands17** [comment, marked as answer]
Hey @aakashkag :wave: 
Currently only the `set` method is supported in Prisma so for now if you have to insert at the end of `child_ids` you would have to do something like this:

```ts
  const res = await prisma.content.findOne({
    where: { id: 1 },
    select: {
      child_ids: true,
    },
  })
  await prisma.content.update({
    data: {
      child_ids: {
        set: [...res.child_ids, 4],
      },
    },
    where: { id: 1 },
  })
```

First you would have to get the ids and then add the value to the end.
The same can be done for inserting at a specific index or deleting.

**Reply by kino1ol** [reply]
> @Bacale-Kevin- Вы также можете использовать `push`. Например,
> 
> ```ts
> data: {
>    push: newAtendee,
> }
> ```
> 
> Это должно натолкнуть `newAtendee`на`booking.attendees`

////////////////////////////////////////////////////////////////////////////////////////

**Now is easier:**
```
await prisma.content.update({
    data: {
      child_ids: {
        push: 4
      },
    },
    where: { id: 1 },
  })
```

**Comment by shamxeed** [comment]
Any progress on this please?

**Comment by felinto-dev** [comment]
I have the same problem. I really would like to know if there has been progressing since then!

**Reply by ryands17** [reply]
Inserting at the end is already available since [2.20.0](https://github.com/prisma/prisma/releases/tag/2.20.0). As for updating at specific index, I would suggest creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that we can look into this.

**Reply by felinto-dev** [reply]
Do you know that when "pushing" an array using Prisma ORM, duplicate values are ignored?

**Reply by ryands17** [reply]
Duplicate values are not ignored as far as I know.

**Reply by Freundschaft** [reply]
how do i push into an array with prisma?

**Reply by ryands17** [reply]
@Freundschaft 👋 
Could you share your schema?

**Reply by Freundschaft** [reply]
I think I have it working now,

schema is:

```prisma
model Booking {
  id            Int         @default(autoincrement()) @id
  uid           String      @unique
  user          User?       @relation(fields: [userId], references: [id])
  userId        Int?
  references    BookingReference[]
  eventType     EventType?  @relation(fields: [eventTypeId], references: [id])
  eventTypeId   Int?

  title         String
  description   String?
  startTime     DateTime
  endTime       DateTime

  attendees     Attendee[]

  createdAt     DateTime    @default(now())
  updatedAt     DateTime?
}
```

and I'm using this code:

```ts
        await prisma.booking.update({
            data: {
                attendees: {
                    create: [{
                        email: req.body.email,
                        name: req.body.name,
                        timeZone: req.body.timeZone
                    }],
                },
            },
            where: {
                id: slotBookingId
            },
        });
```

**Reply by kasir-barati** [reply]
@Freundschaft I am not sure why you said I have it working now. Because I cannot see any `push` or `set`. You just use the `create` operator. So I can conclude that the create push the new Item into the array. Do I understand correctly?

**Reply by Bacale-Kevin** [reply]
any update on this issue please ???

**Reply by raulsanchez1024** [reply]
@Bacale-Kevin - You can also use `push`. E.g.,

```typescript
data: {
   push: newAtendee,
}
```

This should push `newAtendee` into `booking.attendees`

**Reply by Mando-C137** [reply]
@Bacale-Kevin - You should probably use the **push** - api. 
```
const getUser = await prisma.user.update({
  where: {
    id: 9
  },
  data: {
    posts: {
      connect: {
        id: 11
      }
    }
  }
})
```
This just adds the post with id 11 to the user or throws an error if the user does not exist(https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-an-existing-record).

**Reply by danstacyalltech** [reply]
I am having a similar issue. I am successfully able to create an activity in my Mongo db, but the activity is not saving to the user's activity array.

Here is my Prisma schema:
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model User {
  id    String @id @default(auto()) @map("_id") @db.ObjectId
  email String @unique
  name  String?
  avatar String? @default("")
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  activities Activity[] 
  @@map("users")
}

model Activity {
  id    String @id @default(auto()) @map("_id") @db.ObjectId
  // date DateTime
  type String
  distance String
  time String
  region String
  description String?
  user User @relation(fields: [userId], references: [id])
  userId String @db.ObjectId
  @@map("activities")
}
```

Here is my Next.js 14 server action function:

```
export const createActivityAction = async (
  type: string,
  distance: string,
  time: string,
  region: string,
  userId: string,
  description: string
) => {
  try {
    const activity = await prisma.activity.create({
      data: {
        type,
        distance,
        time,
        region,
        description,
        user: {
          connect: {
            id: userId,
          },
        },
      },
    });

    const user = await prisma.user.update({
      where: {
        id: userId,
      },
      data: {
        activities: {
          connect: {
            id: activity.id,
          },
        },
      },
    });
    revalidatePath("/create-activity");
    return { activity, user };
  } catch (error) {
    return { message: "Database Error:Failed to Create Post" };
  }
};
```

**Comment by Apestein** [comment]
push does not work with mySQL, here is the solution
``` 
      const status = await ctx.prisma.post.update({
        where: {
          id: input.postId,
        },
        data: {
          likedBy: {
            connect: { id: userId },
          },
        },
      })
 ```

**Comment by ludralph** [comment]
Hi @aakashkag,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
